snuba/clickhouse/columns.py

```diff
@@ -190,6 +191,36 @@ def for_schema(self) -> str:
         return "{} DEFAULT {}".format(self.inner_type.for_schema(), self.default)
 
 
+class ReadOnly(ColumnType):
```

Comment:
Are these ReadOnly and Nullable the only two modifiers you except to support? 
Are Materialized columns going to be present in the schema?

Reply:
ReadOnly is supposed to be the way that materialized columns are represented in the schema. I plan to move Materialized along with the rest of ColumnTypeWithModifier into the migrations module since they won't be used in storage schemas anymore.

Reply:
ok.

I would suggest you also think about an additional requirement for the schema (that you may or may not already have inmind). The schema should also be usable by the processor to. decide which fields not to fill. Time ago we were considering automating the generation of the processor starting from a simple mapping between event schema and entity schema. 
I guess that readOnly and Nullable may be enough, but just food for thoughts.

Reply:
Yes, I introduced ReadOnly since it's useful for the replacer, but we could use it in the processor as well. We also do pass a list of required fields to the replacer, and I wonder if "required" should be part of the schema instead since it may also be useful for the processor and more easily shared.